services/web-app/pages/_middleware.js

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright IBM Corp. 2022, 2022
+ *
+ * This source code is licensed under the Apache-2.0 license found in the
+ * LICENSE file in the root directory of this source tree.
+ */
+
+import { NextResponse } from 'next/server'
+
+import { isValidIbmEmail } from '@/utils/string'
+
+import { getRunMode, RunMode } from '../../../packages/api/runtime'
+
+/**
+ * Creates request options for retrieveUser api call including headers and agent if necessary
+ *
+ * @param {NextRequest} req getServerSideProps request object
+ * @returns {RequestInit} request options
+ */
+function getRequestOptions(req) {
+  let cookies = ''
+  Object.entries(req.cookies).forEach(([key, val]) => {
+    cookies += `${key}=${val}; `
```

Comment:
Rather than building this value up by hand, let's instead use [cookie](https://www.npmjs.com/package/cookie). Or extract only the cookie value we know we should have, which would be the session id.